packages/tupaia-web/src/utils/useModal.ts

```diff
@@ -3,18 +3,26 @@
  *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
  */
 
-import { useLocation, useNavigate } from 'react-router-dom';
-import { MODAL_ROUTES } from '../constants';
+import { useLocation, useNavigate, useSearchParams } from 'react-router-dom';
+import { MODAL_ROUTES, PASSWORD_RESET_TOKEN_PARAM } from '../constants';
 
 export const useModal = () => {
   const navigate = useNavigate();
   const { hash, ...location } = useLocation();
+  const [urlParams] = useSearchParams();
 
   function navigateToModal(hashKey: `${MODAL_ROUTES}`) {
     navigate({ ...location, hash: hashKey });
   }
   function closeModal() {
-    navigate(location);
+    // Remove the password reset token from the url, if it is set
+    if (urlParams.get(PASSWORD_RESET_TOKEN_PARAM)) {
```

Comment:
It would be nice to keep the specific url params out of the util if possible to prevent it getting messy. Can remove this from here and add the `urlParams.delete(PASSWORD_RESET_TOKEN_PARAM)` to the `useResetPassword` `onSuccess` handler. That would also clear the url params after the form submits instead of when the modal is closed which is more appropriate any way.